### PR TITLE
Container test

### PR DESCRIPTION
**Pull request by isacandrei**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,32 +25,34 @@
...
     "clean": [
       "@stop",
-      "cd dockyard && docker-compose down",
+      "cd dockyard_testing && docker-compose down",
       "cd ..",
-      "rm -rf dockyard/*"
+      "rm -rf dockyard_testing/*"
     ]
   }
 }
```
Comment (nibra) [inline composer.json]: Why did you rename `dockyard`? It is already for testing only. The composer commands are ment to support development of the Virtualisation tool, for explorative testing of the generated setup.

File: src/ServerConfig.php [hunk 2]
```diff
@@ -61,6 +61,12 @@ class ServerConfig
 		'server.type'         => "nginx",
 		'session.handler'     => "database",
 		'session.lifetime'    => "15",
+		'network.name'		  => "joomla",
+		'selenium.no'		  => "1",
+		'selenium.version'	  => "latest",
+		'weblinks.repoOwner'  => "joomla",
+		'weblinks.repoName'   => "weblinks",
+		'weblinks.repoBranch' => "container-test",
```
Comment (nibra) [inline src/ServerConfig.php]: Better use `extension.repoOwner`, `extension.repoName`, and `extension.repoBranch` to keep it independent from WebLinks..

File: src/Service/Apache.php [hunk 3]
```diff
@@ -40,21 +42,20 @@ public function getSetup()
...
 
-		return array_merge(
-			$this->setup,
-			parent::getSetup()
-		);
+//		return array_merge(
+//			$this->setup,
+//			parent::getSetup()
+//		);
+		return $this->setup;
```
Comment (nibra) [inline src/Service/Apache.php:58]: How about the parent (PHP) setup?

File: tests/backup/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    networks:
+      - joomla
+  client:
+    image: 'joomlaprojects/joomla-testing-client-firefox:2.53.1'
+    volumes:
+      - '/Users/isac/Documents/Work/GSOC/joomla/weblinks:/usr/src/tests'
+    networks:
+      - joomla
+    ports:
+      - '5901:5900'
```
Comment (nibra) [inline tests/backup/docker-compose.yml]: This file does not belong to the repo...

File: src/Service/Apache.php (hunk 3 above)
Comment (isacandrei) [reply src/Service/Apache.php:58]: The PhpBase abstract class had no implementation for getSetup(). There main reasons for which I have removed the parent::getSetup() is because it would throw an error if no selenium container would be attached to the apache container: https://github.com/isacandrei/virtualisation/blob/container-test/src/ServiceFactory.php#L47

I believe the virtualisation scope is to have one selenium container for each server, but in the parallel testing platform, we wish to have a configurable number of selenium containers.

I have left the old code there because I thought this may not be ideal, or future proof, that's why I asking for a better solution. Is it a good idea to check here(https://github.com/isacandrei/virtualisation/blob/container-test/src/Service/AbstractService.php#L66-L75) if `services` is empty?

File: tests/backup/docker-compose.yml (hunk 4 above)
Comment (isacandrei) [reply tests/backup/docker-compose.yml]: removed.

File: src/ServerConfig.php (hunk 2 above)
Comment (isacandrei) [reply src/ServerConfig.php]: After discussion with Tito, the extension will be cloned in the parent repository (main coordinator) and only the path of the extension will be used in the config.

File: composer.json (hunk 1 above)
Comment (isacandrei) [reply composer.json]: changed back :)
